logic/src/commonMain/kotlin/com/wire/kalium/logic/sync/receiver/FeatureConfigEventReceiver.kt

```diff
@@ -161,22 +162,24 @@ internal class FeatureConfigEventReceiverImpl internal constructor(
         } else {
             val (currentIsSelfDeletingMessagesEnabled, currentEnforcedTimeout) = userConfigRepository
                 .getSelfDeletingMessagesStatus()
-                .fold({ true to null }, { it.isEnabled to it.enforcedTimeoutInSeconds })
+                .fold({ true to null }, { it.isFeatureEnabled to it.globalSelfDeletionDuration })
 
             when (status) {
                 Status.ENABLED -> userConfigRepository.setSelfDeletingMessagesStatus(
                     SelfDeletingMessagesStatus(
-                        isEnabled = true,
-                        isStatusChanged = !currentIsSelfDeletingMessagesEnabled || currentEnforcedTimeout != enforcedTimeoutSeconds,
-                        enforcedTimeoutInSeconds = if (enforcedTimeoutSeconds == 0) null else enforcedTimeoutSeconds
+                        isFeatureEnabled = true,
+                        hasFeatureChanged = !currentIsSelfDeletingMessagesEnabled || currentEnforcedTimeout != enforcedTimeoutSeconds,
+                        globalSelfDeletionDuration = if (enforcedTimeoutSeconds == 0) null else enforcedTimeoutSeconds,
+                        isEnforced = enforcedTimeoutSeconds.isGreaterThan(0)
                     )
                 )
 
                 Status.DISABLED -> userConfigRepository.setSelfDeletingMessagesStatus(
                     SelfDeletingMessagesStatus(
-                        isEnabled = false,
-                        isStatusChanged = currentIsSelfDeletingMessagesEnabled,
-                        enforcedTimeoutInSeconds = enforcedTimeoutSeconds
+                        isFeatureEnabled = false,
+                        hasFeatureChanged = currentIsSelfDeletingMessagesEnabled,
+                        globalSelfDeletionDuration = enforcedTimeoutSeconds,
+                        isEnforced = false
```

Comment:
when the status is enabled we could be setting a null to globalSelfDeletionDuration but when it is disabled we are not ? abit  confusing 🤔